test/Sign.Core.Test/Native/SignedCmiManifest2Tests.cs

```diff
@@ -0,0 +1,100 @@
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+using System.Xml;
+
+namespace Sign.Core.Test
+{
+    [Collection(SigningTestsCollection.Name)]
+    public class SignedCmiManifest2Tests
+    {
+        private readonly CertificatesFixture _certificatesFixture;
+
+        public SignedCmiManifest2Tests(CertificatesFixture certificatesFixture)
+        {
+            ArgumentNullException.ThrowIfNull(certificatesFixture, nameof(certificatesFixture));
+
+            _certificatesFixture = certificatesFixture;
+        }
+
+        [Fact]
+        [SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "<Pending>")]
+        public void Sign_Never_GeneratesSha1MessageImprint()
+        {
+            using (X509Certificate2 certificate = CreateCertificate())
+            using (RSA privateKey = certificate.GetRSAPrivateKey()!)
+            {
+                XmlDocument manifest = new() { PreserveWhitespace = true };
+
+                using (StringReader reader = new(@$"<?xml version=""1.0"" encoding=""utf-8""?>
+    <asmv1:assembly xsi:schemaLocation=""urn:schemas-microsoft-com:asm.v1 assembly.adaptive.xsd"" manifestVersion=""1.0"" xmlns:asmv1=""urn:schemas-microsoft-com:asm.v1"" xmlns=""urn:schemas-microsoft-com:asm.v2"" xmlns:asmv2=""urn:schemas-microsoft-com:asm.v2"" xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns:co.v1=""urn:schemas-microsoft-com:clickonce.v1"" xmlns:asmv3=""urn:schemas-microsoft-com:asm.v3"" xmlns:dsig=""http://www.w3.org/2000/09/xmldsig#"" xmlns:co.v2=""urn:schemas-microsoft-com:clickonce.v2"">
+      <assemblyIdentity name=""WinFormsApp.application"" version=""1.0.0.0"" publicKeyToken=""46deb9a9283e4567"" language=""neutral"" processorArchitecture=""msil"" xmlns=""urn:schemas-microsoft-com:asm.v1"" />
+      <publisherIdentity name=""{certificate.Subject}"" />
+    </asmv1:assembly>"))
+                {
+                    manifest.Load(reader);
+                }
+
+                SignedCmiManifest2 signedCmiManifest = new(manifest);
+                CmiManifestSigner2 signer = new(privateKey, certificate)
+                {
+                    Flag = CmiManifestSignerFlag.DontReplacePublicKeyToken
+                };
+
+                using (TemporaryEnvironmentPath temporaryEnvironmentPath = CreateTemporaryEnvironmentPath())
+                {
+                    signedCmiManifest.Sign(signer, _certificatesFixture.TimestampServiceUrl.AbsoluteUri);
+                }
+
+                byte[] bytes = GetTimestampBytes(manifest);
+
+                Assert.True(
+                    Rfc3161TimestampToken.TryDecode(
+                        bytes,
+                        out Rfc3161TimestampToken? timestampToken,
+                        out int bytesConsumed));
+                Assert.True(timestampToken.TokenInfo.HashAlgorithmId.IsEqualTo(Oids.Sha256));
```

Comment:
`Assert.NotNull(timestampToken)` above this line?

Reply:
Not necessary.  `Rfc3161TimestampToken.TryDecode(...)` has `[NotNullWhen(true)]` on the `Rfc3161TimestampToken` parameter, so it should never be `null` by line 62, and I'm not trying to test .NET itself.